tests/cases/standalone/common/alter/alter_table_alter_column_set_default.result

```diff
@@ -0,0 +1,103 @@
+|       |   "l" INT NULL DEFAULT 201,                                 |
+|       |   "m" TIMESTAMP(3) NULL DEFAULT '2024-01-30 00:00:00+0000', |
+|       |   "n" INT NULL DEFAULT 300,                                 |
+|       |   TIME INDEX ("j"),                                         |
+|       |   PRIMARY KEY ("i")                                         |
+|       | )                                                           |
+|       |                                                             |
+|       | ENGINE=mito                                                 |
+|       |                                                             |
++-------+-------------------------------------------------------------+
+
+INSERT INTO test1 VALUES (1, '2024-01-30 12:00:00', DEFAULT, DEFAULT, DEFAULT, DEFAULT);
+
+Affected Rows: 1
+
+SELECT * FROM test1;
+
++---+---------------------+-----+-----+---------------------+-----+
+| i | j                   | k   | l   | m                   | n   |
++---+---------------------+-----+-----+---------------------+-----+
+| 1 | 2024-01-30T12:00:00 | 101 | 201 | 2024-01-30T00:00:00 | 300 |
++---+---------------------+-----+-----+---------------------+-----+
+
+ALTER TABLE test1 ADD COLUMN o INTEGER NOT NULL DEFAULT 400;
+
+Affected Rows: 0
+
+ALTER TABLE test1 MODIFY COLUMN o SET DEFAULT 401;
+
+Affected Rows: 0
+
+SHOW CREATE TABLE test1;
+
++-------+-------------------------------------------------------------+
+| Table | Create Table                                                |
++-------+-------------------------------------------------------------+
+| test1 | CREATE TABLE IF NOT EXISTS "test1" (                        |
+|       |   "i" INT NULL,                                             |
+|       |   "j" TIMESTAMP(3) NOT NULL,                                |
+|       |   "k" INT NULL DEFAULT 101,                                 |
+|       |   "l" INT NULL DEFAULT 201,                                 |
+|       |   "m" TIMESTAMP(3) NULL DEFAULT '2024-01-30 00:00:00+0000', |
+|       |   "n" INT NULL DEFAULT 300,                                 |
+|       |   "o" INT NOT NULL DEFAULT 401,                             |
+|       |   TIME INDEX ("j"),                                         |
+|       |   PRIMARY KEY ("i")                                         |
+|       | )                                                           |
+|       |                                                             |
+|       | ENGINE=mito                                                 |
+|       |                                                             |
++-------+-------------------------------------------------------------+
+
+ALTER TABLE test1 MODIFY COLUMN o SET DEFAULT "not allow";
+
+Error: 1001(Unsupported), Unsupported expr in default constraint: Identifier(Ident { value: "not allow", quote_style: Some('"'), span: Span(Location(1,47)..Location(1,58)) }) for column: o
```

Comment:
Since `Expr` supports Display, we can modify the `snafu(display)` of the error to print expr in display mode.

```rust
    #[snafu(display( 
         "Unsupported expr in default constraint: {} for column: {}", 
         expr, 
         column_name 
     ))] 
```

https://github.com/GreptimeTeam/greptimedb/blob/fac8c3e62cf0f3f08ce15cea072198165fef9fd6/src/common/sql/src/error.rs#L57-L67